bindings/rust/src/common.rs

```diff
@@ -0,0 +1,76 @@
+use std::ffi::CString;
+use crate::c_bindings;
+
+#[allow(dead_code)]
+pub struct DeviceDefinition {
+    pub def: c_bindings::InputtinoDeviceDefinition,
+    // Keep those around since we are passing them as pointers
+    name: CString,
+    phys: CString,
+    uniq: CString,
+}
+
+impl DeviceDefinition {
+    pub fn new(name: &str, vendor_id: u16, product_id: u16, version: u16, phys: &str, uniq: &str) -> Self {
```

Comment:
Public functions and types should have docstrings.

Also, you can use `impl AsRef<str>` and `impl Into<u16>` to give callers more flexibility. For example, that would allow you to use the `evdev` enums without converting to a raw u16 first.